plugins/generator-1.21.1/neoforge-1.21.1/templates/block/block.java.ftl

```diff
@@ -349,10 +360,14 @@ public class ${name}Block extends
 
 	<#macro initCustomBlockStateProperties>
 		<#list data.customProperties as prop>
-			<#assign propName = prop.property().getName().replace("CUSTOM:", "")>
-			.setValue(${propName?upper_case},
+			<#assign propName = prop.property().getName()>
+			.setValue(${propName.replace("CUSTOM:", "")?upper_case},
 				<#if prop.property().getClass().getSimpleName().equals("StringType")>
-				${StringUtils.snakeToCamel(propName)}Property.${prop.value()?upper_case}
+					<#if propName.startsWith("CUSTOM:")>
+					${StringUtils.snakeToCamel(propName.replace("CUSTOM:", ""))}Property.${prop.value()?upper_case}
+					<#else>
+					(${generator.map(propName, "blockstateproperties", 2)}) ${propName?upper_case}.getValue("${prop.value()}").get()

```

Comment:
I don't think a cast here is needed:

![image](https://github.com/user-attachments/assets/0633e361-8348-46c3-a54c-4c2a7860119e)

Reply:
Applies to both MC versions

Reply:
:heavy_check_mark: Fixed